testworld/missing_node_test.go

```diff
@@ -0,0 +1,87 @@
+// +build testworld
+
+package testworld
+
+import (
+	"context"
+	"math/big"
+	"net/http"
+	"testing"
+
+	"github.com/centrifuge/go-centrifuge/config"
+	"github.com/centrifuge/go-centrifuge/config/configstore"
+	"github.com/centrifuge/go-centrifuge/contextutil"
+	"github.com/centrifuge/go-centrifuge/identity"
+	"github.com/centrifuge/go-centrifuge/testingutils/identity"
+	"github.com/centrifuge/go-centrifuge/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMissingNode_InvalidIdentity(t *testing.T) {
+	t.Parallel()
+
+	// Hosts
+	alice := doctorFord.getHostTestSuite(t, "Alice")
+
+	// RandomDID
+	randomDID := testingidentity.GenerateRandomDID()
+
+	// Alice shares a document with a randomly generated DID
+	res := createDocument(alice.httpExpect, alice.id.String(), typeInvoice, http.StatusOK, defaultInvoicePayload([]string{randomDID.String()}))
+	txID := getTransactionID(t, res)
+	status, message := getTransactionStatusAndMessage(alice.httpExpect, alice.id.String(), txID)
+	if status != "failed" {
+		t.Error(message)
+	}
+
+	errorMessage := "failed to send document to the node: bytecode for deployed identity contract " + randomDID.String() + " not correct"
+	assert.Contains(t, message, errorMessage)
+}
+
+func TestMissingNode_MissingP2PKey(t *testing.T) {
+	t.Parallel()
+
+	// Hosts
+	alice := doctorFord.getHostTestSuite(t, "Alice")
+
+	// RandomDID without P2P Discovery Key
+	randomDID := createIdentity(t, alice.host.idFactory, alice.host.idService, alice.host.config)
+
+	// Alice shares a document with a randomly generated DID with missing P2P Key
+	res := createDocument(alice.httpExpect, alice.id.String(), typeInvoice, http.StatusOK, defaultInvoicePayload([]string{randomDID.String()}))
+	txID := getTransactionID(t, res)
+	status, message := getTransactionStatusAndMessage(alice.httpExpect, alice.id.String(), txID)
+	if status != "failed" {
+		t.Error(message)
+	}
+
+	assert.Contains(t, message, "missing p2p key")
```

Comment:
As explained on DM lets add the correct error string here and also a check for the anchor on chain. 
Thank you 👍